File: sorting/cxx/selection_sort.cxx
#include <algorithm>
#include <vector>

void selection_sort(std::vector<int>& array)
{
	int minIdx;
	for(int i = 0; i < array.size(); i++)
	{
		minIdx = i;
		for(int j = i + 1; j < array.size(); j++)
			if(array[j] < array[minIdx])
				minIdx = j;
		std::swap(array[i], array[minIdx]);
	}
}

File: sorting/py3/selection_sort.py
# Time:  O(n^2)
# Space: O(1)

def selection_sort(array):
    for i in range(len(array)):
        minIdx = i
        for j in range(i + 1, len(array)):
            if array[j] < array[minIdx]:
                minIdx = j
        array[i], array[minIdx] = array[minIdx], array[i]
